### ISSUE TITLE
cdk-assets: Improve error message when docker is not installed and CDK_DOCKER is not set. 

### ISSUE THREAD

**Issue by vsiravar** [aws/aws-cdk#25657]
### Describe the bug

When docker is not installed and a user wants to use a drop in replacement, if the user does not set CDK_DOCKER=<drop in replacement>, the error message says "Unable to execute 'docker' in order to build a container asset. Please install 'docker' and try again." https://github.com/aws/aws-cdk/blob/main/packages/cdk-assets/lib/private/docker.ts#L186

### Expected Behavior

Error message should inform users about CDK_DOCKER environment variable which can be set to support drop in replacements for docker. 

### Current Behavior

Error message is very docker specific. "Unable to execute 'docker' in order to build a container asset. Please install 'docker' and try again." 

### Reproduction Steps

1. Uninstall docker. 
2. Clone https://github.com/aws/aws-cdk/
3. yarn install
4. NODE_OPTIONS=--max-old-space-size=8192 npx lerna run build --scope=aws-cdk-lib --scope=@aws-cdk-testing/framework-integ
5. cd packages/@aws-cdk-testing/framework-integ/test/aws-lambda/test && node integ.bundling.js

### CDK CLI Version

2.79.1 (build 2e7f8b7)

### Node.js Version

v18.16.0

### OS

macOS 13.3.1

### Language

Typescript

**Comment by pahud** [comment]
Yes we should add CDK_DOCKER in our document or [contributing guide](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md). But at this moment, according to the [Setup](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md#setup) of the contributing guide, **Docker >=19.03** with the running daemon is required.

**Comment by pahud** [comment]
related to https://github.com/aws/aws-cdk/issues/23680#issuecomment-1406516147

**Comment by vsiravar** [comment]
Makes sense, should it additionally be added to the error message as well, informing the user to either set CDK_DOCKER or have a docker daemon running when either of the dependencies are not configured?

**Comment by athewsey** [comment]
Any progress on this? #23680 was closed as completed in Jan, but I keep having customers ask (today also) whether it's possible to use CDK features like [PythonFunction](https://docs.aws.amazon.com/cdk/api/v2/docs/@aws-cdk_aws-lambda-python-alpha.PythonFunction.html), [DockerImageFunction](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_lambda.DockerImageFunction.html), [DockerImageAsset](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_ecr_assets.DockerImageAsset.html) with alternative container build/runtimes - because the documentation for these features is so Docker-oriented and the availability of other options feels more like secret magic than a supported solution that Docker-constrained users can find out about via self-service.
